x-pack/filebeat/input/azureblobstorage/client.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureblobstorage
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/types"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// fetchServiceClientAndCreds , generic function that returns a ServiceClient instance and a ServiceCredentials instance
+func fetchServiceClientAndCreds(cfg config, url string, log *logp.Logger) (*azblob.ServiceClient, *types.ServiceCredentials, error) {
+	if cfg.Auth.SharedCredentials != nil {
```

Comment:
A switch would be clearer.

Reply:
sure

Reply:
we cannot switch on different types here .. if you see , we are 1st checking **cfg.Auth.SharedCredentials** , then on **cfg.Auth.SharedCredentials** , the type has not yet been determined.